primitives/src/merkle_tree/field_based_mht/batch_mht/mod.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{BatchFieldBasedHash};
+
+pub mod poseidon;
+
+pub trait BatchMerkleTree:Clone {
+    type Hash: BatchFieldBasedHash;
+
+    // This function reset the pointing indices for the new element positions and processed positions
+    // Mainly used for benchmarking
+    fn reset(&mut self);
+
+    // This function adds a leaf to the Merkle tree
+    // Whenever the number of added leafs reaches the block size passed as parameter during creation of the
```

Comment:
This should be not known from the interface. The user only needs to know that this function adds a leaf to the Merkle Tree and that there is a hidden logic that handles everything. You also mention that the user will pass the block size during creation of the tree, but this interface doesn't know anything about how the tree will be created. Probably the comment can just be: "Append a new leaf to the merkle tree and updates the other nodes accordingly" or something like this